samples/net/aws_iot/src/main.c

```diff
@@ -0,0 +1,402 @@
+
+	/* If modem FOTA has been carried out, the modem needs to be reinitialized.
+	 * This is carried out by bringing the network interface down/up.
+	 */
+	if (evt->data.image & DFU_TARGET_IMAGE_TYPE_ANY_MODEM) {
+		LOG_INF("Modem FOTA done, reinitializing the modem");
+
+		err = conn_mgr_all_if_down(true);
+		if (err) {
+			LOG_ERR("conn_mgr_all_if_down, error: %d", err);
+			FATAL_ERROR();
+			return;
+		}
+
+		err = conn_mgr_all_if_up(true);
+		if (err) {
+			LOG_ERR("conn_mgr_all_if_up, error: %d", err);
+			FATAL_ERROR();
+			return;
+		}
+
+	} else if (evt->data.image & DFU_TARGET_IMAGE_TYPE_ANY_APPLICATION) {
+		LOG_INF("Application FOTA done, rebooting");
+		IF_ENABLED(CONFIG_REBOOT, (sys_reboot(0)));
+	} else {
+		LOG_WRN("Unexpected FOTA image type");
+	}
+}
+
+static void on_net_event_l4_connected(void)
+{
+	(void)k_work_reschedule(&connect_work, K_SECONDS(5));
+}
+
+static void on_net_event_l4_disconnected(void)
+{
+	(void)aws_iot_disconnect();
+	(void)k_work_cancel_delayable(&connect_work);
+	(void)k_work_cancel_delayable(&shadow_update_work);
+}
+
+/* Event handlers */
+
+static void aws_iot_event_handler(const struct aws_iot_evt *const evt)
+{
+	switch (evt->type) {
+	case AWS_IOT_EVT_CONNECTING:
+		LOG_INF("AWS_IOT_EVT_CONNECTING");
+		break;
+	case AWS_IOT_EVT_CONNECTED:
+		LOG_INF("AWS_IOT_EVT_CONNECTED");
+		on_aws_iot_evt_connected(evt);
+		break;
+	case AWS_IOT_EVT_READY:
+		LOG_INF("AWS_IOT_EVT_READY");
+		break;
+	case AWS_IOT_EVT_DISCONNECTED:
+		LOG_INF("AWS_IOT_EVT_DISCONNECTED");
+		on_aws_iot_evt_disconnected();
+		break;
+	case AWS_IOT_EVT_DATA_RECEIVED:
+		LOG_INF("AWS_IOT_EVT_DATA_RECEIVED");
+
+		LOG_INF("Received message: \"%.*s\" on topic: \"%.*s\"", evt->data.msg.len,
+									 evt->data.msg.ptr,
+									 evt->data.msg.topic.len,
+									 evt->data.msg.topic.str);
+		break;
+	case AWS_IOT_EVT_PUBACK:
+		LOG_INF("AWS_IOT_EVT_PUBACK, message ID: %d", evt->data.message_id);
+		break;
+	case AWS_IOT_EVT_PINGRESP:
+		LOG_INF("AWS_IOT_EVT_PINGRESP");
+		break;
+	case AWS_IOT_EVT_FOTA_START:
+		LOG_INF("AWS_IOT_EVT_FOTA_START");
+		break;
+	case AWS_IOT_EVT_FOTA_ERASE_PENDING:
+		LOG_INF("AWS_IOT_EVT_FOTA_ERASE_PENDING");
+		break;
+	case AWS_IOT_EVT_FOTA_ERASE_DONE:
+		LOG_INF("AWS_FOTA_EVT_ERASE_DONE");
+		break;
+	case AWS_IOT_EVT_FOTA_DONE:
+		LOG_INF("AWS_IOT_EVT_FOTA_DONE");
+		on_aws_iot_evt_fota_done(evt);
+		break;
+	case AWS_IOT_EVT_FOTA_DL_PROGRESS:
+		LOG_INF("AWS_IOT_EVT_FOTA_DL_PROGRESS, (%d%%)", evt->data.fota_progress);
+		break;
+	case AWS_IOT_EVT_ERROR:
+		LOG_INF("AWS_IOT_EVT_ERROR, %d", evt->data.err);
+		break;
+	case AWS_IOT_EVT_FOTA_ERROR:
+		LOG_INF("AWS_IOT_EVT_FOTA_ERROR");
+		break;
+	default:
+		LOG_WRN("Unknown AWS IoT event type: %d", evt->type);
+		break;
+	}
+}
+
+static void l4_event_handler(struct net_mgmt_event_callback *cb,
+			     uint32_t event,
+			     struct net_if *iface)
+{
+	switch (event) {
+	case NET_EVENT_L4_CONNECTED:
+		LOG_INF("IP Up");
```

Comment:
If you're going to push any more changes to the PR, you can include some more descriptive logging in this handler. It's a little cryptic as it is.

Reply:
Will do